### DISCUSSION TITLE
Refresh SSR page does not fetch data

### DISCUSSION THREAD

**Discussion by vionijannet** [nuxt/nuxt#21647] [Questions]
I have this detail page 
I'm trying to fetch data using useFetch, but here's the problem

```
// index.vue (List)
<template>
    <NuxtLink to="/abc">Link</NuxtLink>
<template>

// [id].vue (Detail)
<template>
    {{ data }}
</template>

<script setup lang="ts">
const config = useRuntimeConfig();
const { data } = useFetch(config.public.api, {
    server: true,
});
```

If I open detail page from list, it can fetch data
But if I refresh the detail page, it doesn't fetch data
Is there anything wrong in my code? 
Thanks in advance

**Comment by davidwwu** [comment]
I'm having the same issue, where navigating from another page works but is empty when refreshed.

**Reply by xjccc** [reply]
await useFetch?

**Reply by davidwwu** [reply]
Yes, my useFetch calls are in stores (from Pinia) and I tried calling the actions in the components both in `onMounted()` and outside.
When calling in `onMounted()` the call was made from the Nuxt server so I don't know how to see more detail other than printing the result after `await` which is null. When calling outside (at the root of the script tag), it's called from the front end. The console logs null right after the `await useFetch` but the response in the network tab of the console shows the data.
Navigating from another page to the page in question works fine.

**Reply by xjccc** [reply]
`onMounted` is client fetch, you can
// client
```
onMounted (async () => {
  await nextTick()
  useFetch('')
})
```
// server
```
<script setup>
   await useFetch('')
</script>
```

**Reply by davidwwu** [reply]
OMG `await nextTick()` solved my problem! Thanks so much

**Reply by vionijannet** [reply]
Isn't using nextTick will only trigger in client side? I need it work on server side.

**Reply by davidwwu** [reply]
Yes, I need to trigger the useFetch on the client side ('cause we had to use localStorage with pinia-plugin-persistedstate). Calling `useFetch()` like @xjccc mentioned works (i.e., outside of `onMount()`) in other places.
```js
<script setup>
   await useFetch('')
</script>
```

Edit: I think your issue might be this: https://stackoverflow.com/questions/74759799/nuxt-3-ssr-usefetch-returning-null-on-page-refresh, ppl mentioning it could be related to Node version, so look into that too
